File: feign-core/src/main/java/name/abhijitsarkar/feign/core/model/ResponseProperties.java
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * License for more details.
 */

package name.abhijitsarkar.feign.core.model;

import lombok.Getter;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

import static org.springframework.http.HttpStatus.OK;

/**
 * @author Abhijit Sarkar
 */
@Getter
@SuppressWarnings({"PMD.BeanMembersShouldSerialize", "PMD.SingularField"})
public class ResponseProperties {
    private Integer status;
    private Map<String, String> headers;
    private Body body;
    @Setter
    private Delay delay;

    public ResponseProperties() {
        setStatus(null);
        setHeaders(null);
        setBody(null);
    }

    public void setStatus(Integer status) {
        this.status = (status == null) ? Integer.valueOf(OK.value()) : status;
    }

    public void setHeaders(Map<String, String> headers) {
        this.headers = (headers == null) ? new HashMap<>() : headers;
    }

    public void setBody(Body body) {
        this.body = (body == null) ? new Body() : body;
    }
}

File: feign-core/src/main/scala/name/abhijitsarkar/feign/core/model/ResponseProperties.scala
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * A copy of the GNU General Public License accompanies this software,
 * and is also available at http://www.gnu.org/licenses.
 */

package name.abhijitsarkar.feign.core.model

import java.util.{Collections, Map => JavaMap}

import org.springframework.http.HttpStatus.OK

import scala.collection.JavaConverters._

/**
  * @author Abhijit Sarkar
  */
class ResponseProperties {
  var status: Integer = _
  var headers: JavaMap[String, String] = _
  var body: Body = _

  setStatus(null)
  setHeaders(null)
  setBody(null)

  def getStatus = status

  def setStatus(status: Integer) {
    this.status = if (status == null) Integer.valueOf(OK.value)
    else status
  }

  def headersToScala = headers.asScala.toMap

  def getHeaders = headers

  def setHeaders(headers: JavaMap[String, String]) {
    this.headers = if (headers == null) Collections.emptyMap()
    else headers
  }

  def getBody = body

  def setBody(body: Body) {
    this.body = if (body == null) new Body
    else body
  }

  override def toString = s"ResponseProperties($status, $headers, $body)"
}
